### PR TITLE
Use OCR to get Singapore solar data

### PR DESCRIPTION
**Pull request by systemcatch**
Implementation of OCR to get data from the solar map. For this to work the tesseract engine must be able to be invoked by the command 'tesseract' on the shell. I don't know the EM server setup or whether this can be handled by docker but will modify anyway needed. 

ref #653

### CODE REVIEW

File: parsers/SG.py [hunk 1]
```diff
@@ -58,6 +61,41 @@
...
+    # At night format changes from 0.00 to 0
+    # tesseract cannot distinguish singular 0 and O in font provided by image.
+    # This try/except will make sure no invalid data is returned.
+    try:
+        solar = float(val)
+    except ValueError as err:
+        if len(val) == 1 and 'O' in val:
+            solar = 0.0
+        else:
+            print("Singapore solar data is unreadable - got {}.".format(val))
```
Comment (corradio) [inline parsers/SG.py]: maybe raise an exception is in case?

Comment (jarek) [reply parsers/SG.py]: If we raise exception, we lose data from main parser - there is no history backfill in SG

Comment (corradio) [reply parsers/SG.py]: got it

**Comment by systemcatch** [comment]
Right the image is still stuck on the same time. https://github.com/tmrowco/electricitymap/pull/1134/commits/0296bf172b6fa8b4b6423299031369d0f4de404c will make sure that any data older than 1 hour won't be used.

**Comment by jarek** [comment]
We've jinxed it :(

Let's see if it gets fixed when it's Monday in Singapore

**Comment by alixunderplatz** [comment]
The [image](https://www.ema.gov.sg/cmsmedia/irradiance/plot.png) was fixed and is working properly! :)

**Comment by systemcatch** [comment]
@jarek happy with everything?

**Comment by maxbellec** [comment]
loving this!

**Comment by jarek** [comment]
This worked fine today morning for me as I recall. Changes look good, thanks!

**Pull request merged**